sailfishusermanager.cpp

```diff
@@ -290,11 +291,22 @@ uint SailfishUserManager::addUser(const QString &name)
 
 int SailfishUserManager::removeUserFiles(uint uid)
 {
+    int rv = EXIT_FAILURE;
     QDir dir(USER_ENVIRONMENT_DIR.arg(uid));
     if (dir.removeRecursively())
-        return EXIT_SUCCESS;
-    qCWarning(lcSUM) << "Removing user environment directory failed";
-    return EXIT_FAILURE;
+        rv = EXIT_SUCCESS;
+    else
+        qCWarning(lcSUM) << "Removing user environment directory failed";
+
+    // Execute user removal scripts
+    QDir removal(USER_REMOVE_SCRIPT_DIR);
+    for (const QString &entry : removal.entryList(QDir::Files | QDir::Executable)) {
+        int exitCode = QProcess::execute(USER_REMOVE_SCRIPT_DIR + '/' + entry, QStringList() << QString::number(uid));
```

Comment:
Too late etc. More than once bitten: executing everything that exists opens door for weird mishaps via having backup files / package upgrade artefacts getting executed too... enforcing some sort of naming convention (regexp/whatnot) would avoid such things.

Reply:
Also having deterministic execute order never hurts, now it is more or less random due to implicit NoSort.

Reply:
> Too late

Well, I would argue it's too late only once the software has been released (or so, there can be other deadlines as well) so this should be about as easy as creating another PR with the fixes.